src/WebSdk/Publish/Tasks/Tasks/MsDeploy/MSDeploy.cs

```diff
@@ -263,13 +264,24 @@ public string ExePath
 #else
                 if (string.IsNullOrEmpty(m_exePath))
                 {
-                    string programFilesX86 = System.Environment.ExpandEnvironmentVariables("%ProgramFiles(x86)%");
-                    string msdeployExePath = System.IO.Path.Combine("IIS", "Microsoft Web Deploy V3");
-                    m_exePath = IO.Path.Combine(programFilesX86, msdeployExePath);
+                    string programFiles = Environment.ExpandEnvironmentVariables("%ProgramFiles(x86)%");
```

Comment:
I'd have first checked the directory that matches the current process bitness, then the other one. This instead checks the x86 directory first and then x64. But I suppose the directories that msdeploy reads or writes are unlikely to be redirected by wow64 so it won't really matter.

Reply:
If you look at the existing code, it does the x86 check first. Also, this reduces the risk of anyone who might have set the env variable themselves to bypass this issue. 

That is the reason, I left the first portion of the check unchanged. If the file is not present, then it goes through the right checks based on machine and process bitness to find the right exe.